compiler/rustc_middle/src/ty/consts/kind.rs

```diff
@@ -177,6 +178,8 @@ impl<'tcx> ConstKind<'tcx> {
         if let ConstKind::Unevaluated(unevaluated) = self {
```

Comment:
this whole function should be removed, all `ty::ConstKind` should now be evaluated using `try_eval_for_typeck`. (which should then be renamed accordingly)

Reply:
We introduced this to minimize code duplication, as requested by @oli-obk.